### PR TITLE
Add python linter (ruff) for integration test

### PR DESCRIPTION
**Pull request by chhetripradeep**
[Ruff](https://beta.ruff.rs/) has been the state of the art in python linter ecosystem and since we aren't using any, I think we should adopt it for all our python based scripting.

### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 661863129486224f8d564f9ce62a8eaec10bd67b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49869/661863129486224f8d564f9ce62a8eaec10bd67b/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by nikitamikhaylov** [comment]
@chhetripradeep How it differs with Black? https://github.com/psf/black

File: tests/integration/helpers/cluster.py [hunk 1]
```diff
@@ -2326,7 +2326,7 @@ def wait_kafka_is_available(self, kafka_docker_id, kafka_port, max_retries=50):
                 subprocess.check_call(  # STYLE_CHECK_ALLOW_SUBPROCESS_CHECK_CALL
                     self.base_kafka_cmd + ["logs"], stdout=f
                 )
-        except Exception as e:
+        except Exception:
             logging.debug("Unable to get logs from docker.")
```
Comment (tavplubix) [inline tests/integration/helpers/cluster.py:2330]: Let's add `e` to the exception message (seems like it was forgotten)

**Comment by alesapin** [comment]
cc @Felixoid

File: tests/integration/helpers/test_tools.py [hunk 2]
```diff
@@ -25,7 +27,7 @@ def __init__(self, contents):
                 "contents must be either file or string or list, actual type: "
                 + type(contents).__name__
             )
-        self.lines = [l.strip() for l in raw_lines if l.strip()]
+        self.lines = [content.strip() for content in raw_lines if content.strip()]
```
Comment (tavplubix) [inline tests/integration/helpers/test_tools.py:30]: What's wrong with `l`?

**Comment by Felixoid** [comment]
Hello. Thanks for the contribution.

Although, a few points:

1. The place of the check is wrong, it does not belong to the integration tests. We have a style check, and linters belong there.
2. We have [an outdated pylint](https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/style/Dockerfile#L21) there, and it makes sense to update the pylint rather than than bringing a new linter with fewer rules, see the https://github.com/charliermarsh/ruff/issues/689
3. One necessary change that must be in the updated version is this [PR](https://github.com/pylint-dev/pylint/pull/7163). It will allow us to have a central configuration that will work for everybody.

As @tavplubix said, some rules are too strict, e.g. `snake case AKA one-letter-var`, and more.

**Comment by Felixoid** [comment]
> @chhetripradeep How it differs with Black? https://github.com/psf/black

@nikitamikhaylov black is a formatter. Ruff is rather conflicting with pylint that we have

**Comment by chhetripradeep** [comment]
@Felixoid Sorry I didn't know that we're already using pylint for `tests/integration`. I think ruff is redundant then. 
Earlier I noticed pylint call https://github.com/ClickHouse/ClickHouse/blob/master/utils/check-style/check-style#L157-L158 so i thought we aren't using pylint for integration tests

**Comment by Felixoid** [comment]
Strange, we indeed don't check it for integration tests. Let's do it for all `tests/**/*.py`

Just change the mentioned line

**Comment by chhetripradeep** [comment]
Closing this in favour of https://github.com/ClickHouse/ClickHouse/pull/50174

File: tests/integration/helpers/postgres_utility.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def get_postgres_conn(
     database_name="postgres_database",
     replication=False,
 ):
-    if database == True:
+    if database is True:
```
[the remark this answers is not in the document]
Comment (cclauss) [reply tests/integration/helpers/postgres_utility.py:35]: Read PEP8 -- TL;DR Identity is not equality in Python.

File: tests/integration/helpers/test_tools.py (hunk 2 above)
Comment (cclauss) [reply tests/integration/helpers/test_tools.py:30]: Easy to confuse `l` with `1`.

% `ruff rule E741`
# ambiguous-variable-name (E741)

Derived from the **pycodestyle** linter.

## What it does
Checks for the use of the characters 'l', 'O', or 'I' as variable names.

## Why is this bad?
In some fonts, these characters are indistinguishable from the
numerals one and zero. When tempted to use 'l', use 'L' instead.

## Example
```python
l = 0
O = 123
I = 42
```

Use instead:
```python
L = 0
o = 123
i = 42
```

**Pull request closed without merging**